topics/bloomfilters.md

```diff
@@ -0,0 +1,263 @@
+
+Example of default bloom filter information:
+
+```
+127.0.0.1:6379> BF.ADD default_filter item
+1
+127.0.0.1:6379> BF.INFO default_filter
+ 1) Capacity
+ 2) (integer) 100
+ 3) Size
+ 4) (integer) 384
+ 5) Number of filters
+ 6) (integer) 1
+ 7) Number of items inserted
+ 8) (integer) 2
+ 9) Error rate
+10) "0.01"
+11) Expansion rate
+12) (integer) 2
+13) Tightening ratio
+14) "0.5"
+15) Max scaled capacity
+16) (integer) 26214300
+```
+
+## Performance
+
+The bloom commands which involve adding items or checking the existence of items have a time complexity of O(N * K) where N is the number of hash functions used by the bloom filter and K is the number of elements being inserted. This means that both BF.ADD and BF.EXISTS are both O(N) as they only operate on one item.
+
+In case of scalable bloom filters, with every scale out, we increase the number of checks (using hash functions of each sub filter) performed during any add / exists operation. For this reason, it is recommended that users choose a capacity and expansion rate after evaluating the use case / workload to avoid several scale outs and reduce the number of checks.
+
+The other bloom filter commands are O(1) time complexity: BF.CARD, BF.INFO, BF.RESERVE, and BF.INSERT (when no items are provided).
+
+## Monitoring 
+
+To check the server's overall bloom filter metrics, you can use the `INFO BF` or the `INFO MODULES` command. 
+
+Example of `INFO BF` calls in different scenarios:
+
+```
+127.0.0.1:6379> INFO BF
+# bf_bloom_core_metrics
+bf_bloom_total_memory_bytes:0
+bf_bloom_num_objects:0
+bf_bloom_num_filters_across_objects:0
+bf_bloom_num_items_across_objects:0
+bf_bloom_capacity_across_objects:0
+
+# bf_bloom_defrag_metrics
+bf_bloom_defrag_hits:0
+bf_bloom_defrag_misses:0
+127.0.0.1:6379> bf.add key value
+(integer) 1
+127.0.0.1:6379> info bf
+# bf_bloom_core_metrics
+bf_bloom_total_memory_bytes:384
+bf_bloom_num_objects:1
+bf_bloom_num_filters_across_objects:1
+bf_bloom_num_items_across_objects:1
+bf_bloom_capacity_across_objects:100
+
+# bf_bloom_defrag_metrics
+bf_bloom_defrag_hits:0
+bf_bloom_defrag_misses:0
+```
+
+### Bloom filter core metrics
+
+* `bf_bloom_total_memory_bytes`: Current total number of bytes used by all bloom filters.
+
+* `bf_bloom_num_objects`: Current total number of bloom filters.
+
+* `bf_bloom_num_filters_across_objects`: Current total number of sub filters across all bloom filters.
+
+* `bf_bloom_num_items_across_objects`: Current total number of items across all bloom filters.
+
+* `bf_bloom_capacity_across_objects`: Current total capacity across all bloom filters.
+
+### Bloom filter defrag metrics
+
+* `bf_bloom_defrag_hits`: Total number of defrag hits that have occurred on bloom filters.
+
+* `bf_bloom_defrag_misses`: Total number of defrag misses that have occurred on bloom filters.
+
+## Limits
```

Comment:
These are more of configs as opposed to limits.

Reply:
In this document, we do already list the configs in a section above.

How about naming this section as ""Large Bloom Filters"? Or "Handling Large Bloom Filters"?